packages/core/src/lib/components/T/utils/utils.ts

```diff
@@ -0,0 +1,38 @@
+import type { Object3D } from 'three'
+import type { AnyClass, MaybeInstance } from '../types'
+import type { DisposableThreeObject } from '../../../types'
+
+const classRegex = /^\s*class\s+/
+
+// Type Guards
+const isClass = (input: unknown): input is AnyClass => {
+  if (typeof input !== 'function') {
+    return false
+  }
+
+  return classRegex.test(input.toString())
```

Comment:
optimization: cache the regex object. on my machine this showed around a 10x speed improvement for this line!

Reply:
Oh yeah!